src/wfdlgs2.c

```diff
@@ -487,6 +487,40 @@ JAPANEND
                SetDlgItemText(hDlg, IDD_KK_TEXTTO, szStr);
             }
 
+            break;
+         case IDM_HARDLINK:
+
+            p = GetSelection(0, NULL);
+
+            LoadString(hAppInstance, IDS_HARDLINK, szTitle, COUNTOF(szTitle));
+            SetWindowText(hDlg, szTitle);
+
+            if (bJAPAN) {
+
+               //by yutakas 1992/10/29
+               LoadString(hAppInstance, IDS_KK_COPYFROMSTR, szStr, COUNTOF(szStr));

```

Comment:
Japanese resource string `IDS_KK_COPYFROMSTR` means `Copy source` so maybe not so suitable here. This if block seems to be some obsolete code...
Maybe @craigwi knows about the history of JAPANBEGIN and JAPANEND?

Reply:
I know. 
But since all language .rc files contained IDS_KK_COPYFROMSTR I decided to copy the passage in the code.
It looks like as if this means 'From', 'To' and then it is harmless, because it fits for symlink/hardlink creation

Reply:
> I know. But since all language .rc files contained IDS_KK_COPYFROMSTR I decided to copy the passage in the code. It looks like as if this means 'From', 'To' and then it is harmless, because it fits for symlink/hardlink creation

The problem is it means "**copy** from" / "Specify a **copy** source" rather then just `From`... may be some kind of confusing.

Reply:
I understand the coding now. It seems to handle a Japanese specialty:

All languages basically have 'From:', 'To:' (without verb) on this dialog, but Japanese has 'Copy from'<in Japanese> and 'Copy To' <in Japanese> on the dialog (with a verb).

@tokomio-sun : Do we really need a verb? Could we not just only have 'From:', 'To:' without verb in Japanese?

As of this PR I would keep it as it is, even if it says 'Copy from' when it is 'Hardlink from'.
We can cleanup this up in an extra PR.

Reply:
> I understand the coding now. It seems to handle a Japanese specialty:
> 
> All languages basically have 'From:', 'To:' (without verb) on this dialog, but Japanese has 'Copy from' and 'Copy To' on the dialog (with a verb).
> 

I see in `winfile_ja-JP.dlg` that `MOVECOPYDLG`'s `From` is translated as `Move from`, so they do need some code to change the verbs when the dialog is reused for other operations. (...and "removing them" seems to be not so "harmless" ... my mistake)

Although I have no evidence, but I guess `bJAPAN` was something served as a localization method long long ago and may be deprecated nowadays by using multiple `.dlg` and `.rc`s, and I didn't see any code that set `bJAPAN` to `true` after it is given `false` in `winfile.h`... Maybe in another PR we could try to fix this with native developers...

Reply:
Correct it is all about the verb and re-using one dialog for many purposes/verbs
Currently it is up to @tokomio-sun  or @Takym tell us about the necesseity of verbs in this kind of dialog

Reply:
Yes, these need verbs because "元" and "先" are suffixes.
I think "From"/"To" should be translated to "リンク元"/"リンク先" ("リンク" means "link") in `IDM_HARDLINK` and `IDM_SYMLINK`.

Reply:
@Takym : Can you please provide me 4 strings

'hardlink from'
'hardlink to'
'symlink from'
'symlink to'

I will add respective IDS_KK_xxxxSTR to the resources and make the dialogs also work properly beneath if (bJapan)

Reply:
@schinagl I translated all below:

'hardlink from' -> 'ハードリンク元'
'hardlink to' -> 'ハードリンク先'
'symlink from' -> 'シンボリックリンク元'
'symlink to' -> 'シンボリックリンク先'

Reply:
Thx! Added with b678626